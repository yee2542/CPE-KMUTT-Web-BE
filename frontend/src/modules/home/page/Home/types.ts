import { GetHomeQuery } from 'common/generated/generated-types';

export type HomesProps = { data?: GetHomeQuery };
