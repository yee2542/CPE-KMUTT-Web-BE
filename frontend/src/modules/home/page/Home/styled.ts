import styled from 'styled-components';

export const Main = styled.main`
<<<<<<< HEAD:frontend/src/modules/root/page/Home/styled.ts
  padding: 5rem 0;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
=======
  background-color: #fafafa;
>>>>>>> e4db972fd6554c3a342021792462073c4ecf6ab5:frontend/src/modules/home/page/Home/styled.ts
`;

export const Title = styled.h1`
  margin: 0;
  line-height: 1.15;
  font-size: 4rem;
  text-align: center;
  background-color: #fafafa;

  a {
    color: #0070f3;
    text-decoration: none;

    &:hover,
    &:focus,
    &:active {
      text-decoration: underline;
    }
  }
`;
