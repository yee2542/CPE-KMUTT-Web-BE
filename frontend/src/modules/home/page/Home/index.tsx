import React from 'react';

import Head from 'next/head';
import { HiOutlineArrowRight } from 'react-icons/hi';

import { Btn } from 'common/components/Button/styled';
import Carousel from 'common/components/Carousel';
import useIsMobileDisplay from 'common/hooks/useIsMobileDisplay';

import Footer from 'modules/home/components/Footer';
import StaticSection from 'modules/home/components/StaticSection';
import { ButtonStyled } from 'modules/home/components/StaticSection/styled';
import WhatNewSection from 'modules/home/components/WhatNewSection';
import { useHomeContentParser } from 'modules/home/hooks/useHomeContentParser';
import { HomesProps } from 'modules/home/page/Home/types';
import Navbar from 'modules/root/components/Navbar';

import { Main } from './styled';

const Home: React.FC<HomesProps> = ({ data }) => {
  const isMobile = useIsMobileDisplay();
  const { mainCarousal, activityAndAwardsCarousal, whatsNews } = useHomeContentParser(
    data,
  );

  const showForMobile = isMobile;
  const showForDesktop = !isMobile;

  return (
    <div>
      <Head>
        <title>หน้าแรก</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Navbar />
      <Main>
        <Carousel variant="Default" item={mainCarousal} />

        <WhatNewSection data={whatsNews} />

        <StaticSection
          id="1"
          background="/images/static_bg_curriculum.png"
          header="ที่วิศวกรรมคอมพิวเตอร์ มจธ. เรามีหลักสูตรหลากหลาย"
          content="เซ็นเตอร์เวิร์คฟลุทแอดมิสชันออร์แกน ใช้งานรูบิค แซมบ้าเมาท์
          แอดมิชชั่นคัตเอาต์โครนาโอ้ยเซ็นเซอร์ ผู้นำเมจิควิทย์ บลอนด์แทคติคไอซียูเยลลี่">
          <ButtonStyled>
            <Btn $color="primary">
              สำรวจหลักสูตรทั้งหมด
              <HiOutlineArrowRight className="Icon" style={{ marginLeft: '16px' }} />
            </Btn>
          </ButtonStyled>
          <ButtonStyled>
            <Btn $color="white">
              สมัครเข้าเรียน
              <HiOutlineArrowRight
                className="Icon"
                style={{ marginLeft: '16px' }}
                size="16px"
              />
            </Btn>
          </ButtonStyled>
        </StaticSection>
        <Carousel variant="Slider" item={activityAndAwardsCarousal} />
        <StaticSection
          id="2"
          background="/images/static_bg_gallery.png"
          header="แกลเลอรี่รูปภาพ"
          content="โดยมีอาจารย์ผู้สอนเป็นผู้ให้คำปรึกษา นักศึกษาต้องทำงานเป็นกลุ่ม และมีการนำเสนอผลงาน ของตนเองให้เพื่อนๆ ร่วมกันวิเคราะห์และวิจารณ์อย่างมีเหตุผล">
          <ButtonStyled>
            <Btn $color="white">
              สำรวจแกลเลอรี่
              <HiOutlineArrowRight
                className="Icon"
                style={{ marginLeft: '16px' }}
                size="16px"
              />
            </Btn>
          </ButtonStyled>
        </StaticSection>
        <Footer key="1"></Footer>
      </Main>
    </div>
  );
};
export default Home;
