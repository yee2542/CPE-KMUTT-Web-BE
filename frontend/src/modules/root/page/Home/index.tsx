import Container from 'common/components/Container';
import Head from 'next/head';
import React from 'react';
import { Main } from './styled';

const Home: React.FC = () => {
  return (
    <div>
      <Head>
        <title>Computer Engineering | KMUTT</title>
        <link rel="icon" href="/Logo.svg" />
      </Head>
      {/* <Navbar></Navbar> 

          <Header></Header>

          <Announcement></Announcement>

          <HomeCarousel></HomeCarousel>

          <Programme></Programme>

          <Contact></Contact>

          <Footer></Footer>

      */}
      <Container>
        <Main></Main>
      </Container>
    </div>
  );
};

export default Home;
