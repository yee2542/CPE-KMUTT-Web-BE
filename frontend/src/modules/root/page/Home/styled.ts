import styled from 'styled-components';

export const Main = styled.main`
  padding: 5rem 0;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Title = styled.h1`
  margin: 0;
  line-height: 1.15;
  font-size: 4rem;
  text-align: center;
  background-color: #fafafa;

  a {
    color: #0070f3;
    text-decoration: none;

    &:hover,
    &:focus,
    &:active {
      text-decoration: underline;
    }
  }
`;
