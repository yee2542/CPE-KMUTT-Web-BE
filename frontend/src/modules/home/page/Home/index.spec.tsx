import * as React from 'react';

import { shallow } from 'enzyme';

import Home from '.';

describe('Index page', () => {
  it('Index page should be defined', () => {
    const wrap = shallow(<Home />);
    expect(wrap.exists()).toBe(true);
  });

  it('Index page to match snapshot', () => {
    const wrap = shallow(<Home />);
    expect(wrap).toMatchSnapshot();
  });
});
