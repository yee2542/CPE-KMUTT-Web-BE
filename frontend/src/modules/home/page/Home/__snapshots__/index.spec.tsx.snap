// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page Index page to match snapshot 1`] = `
<CommonLayout
  Header={
    <Carousel
      variant="Default"
    />
  }
  showTopBar={false}
>
  <Head>
    <title>
      หน้าแรก
    </title>
    <link
      href="/favicon.ico"
      rel="icon"
    />
  </Head>
  <styled__Main>
    <WhatNewSection />
    <StaticSection
      background="/images/static_bg_curriculum.jpg"
      content="ที่วิศวกรรมคอมพิวเตอร์ มจธ. เรามีหลักสูตรหลากหลาย
    ตอบโจทย์ความต้องการของตลาด ทั้งในไทย และระดับโลก "
      header="หลักสูตร"
      id="1"
    >
      <styled__ButtonStyled>
        <styled__Btn
          $color="primary"
        >
          สำรวจหลักสูตรทั้งหมด
          <HiOutlineArrowRight
            className="Icon"
            style={
              Object {
                "marginLeft": "16px",
              }
            }
          />
        </styled__Btn>
      </styled__ButtonStyled>
      <styled__ButtonStyled>
        <styled__Btn
          $color="white"
        >
          สมัครเข้าเรียน
          <HiOutlineArrowRight
            className="Icon"
            size="16px"
            style={
              Object {
                "marginLeft": "16px",
              }
            }
          />
        </styled__Btn>
      </styled__ButtonStyled>
    </StaticSection>
    <Carousel
      variant="Slider"
    />
    <StaticSection
      background="/images/static_bg_gallery.png"
      content="ประมวลภาพบรรยากาศ กิจกรรมภายในภาควิชาฯ และกิจกรรมภายนอกที่เกี่ยวข้องกับนักศึกษาของภาควิชาฯ"
      header="แกลเลอรี่รูปภาพ"
      id="2"
    />
    <BadgeContactSection />
  </styled__Main>
</CommonLayout>
`;
